compiler/rustc_resolve/src/imports.rs

```diff
@@ -248,6 +248,19 @@ struct UnresolvedImportError {
     candidates: Option<Vec<ImportSuggestion>>,
 }
 
+struct RedundantImportVisibilityMsg {
```

Comment:
I'm not sure why this structure is necessary.
Why cannot the message string be inlined at the use site?
The structure is not used by the internationalization system (`.ftl` stuff) either.